QuoteOfTheDay/Pages/Simulate.cshtml.cs

```diff
@@ -0,0 +1,63 @@
+﻿using Microsoft.ApplicationInsights;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using Microsoft.FeatureManagement;
+using Microsoft.FeatureManagement.FeatureFilters;
+
+namespace QuoteOfTheDay.Pages;
+
+public class SimulateModel : PageModel
+{
+    private readonly IVariantFeatureManager _featureManager;
+    private readonly TelemetryClient _telemetryClient;
+
+    public SimulateModel(IVariantFeatureManager featureManager, TelemetryClient telemetryClient)
+    {
+        _featureManager = featureManager;
+        _telemetryClient = telemetryClient;
+    }
+
+    public async void OnGet()
+    {
+        if (HttpContext.Request.Query?["simulate"].Count > 0) {
+            await SimulateUser();
+        }
+    }
+
+    public async Task SimulateUser() {
+        string randomUserId = Guid.NewGuid().ToString();
+
+        var activityFeature = HttpContext.Features.Get<IHttpActivityFeature>();
+        activityFeature.Activity.SetBaggage("TargetingId", randomUserId);
+
+        Variant variant = await _featureManager.GetVariantAsync("Greeting", new TargetingContext() { UserId= randomUserId });
+
+        // 50% chance to like for None variant
+        if (variant.Name == "None")
```

Comment:
This is dependent upon the variant names. If somebody gets that setup wrong, this seems to do nothing and not warn.

Reply:
Removed simulate for now.